Tasks/KubernetesManifestV0/src/actions/deploy.ts

```diff
@@ -0,0 +1,193 @@
+        });
+    }
+
+    return files;
+}
+
+function replaceAll(currentString: string, replaceToken, replaceValue) {
+    let i = currentString.indexOf(replaceToken);
+    if (i < 0) {
+        return currentString;
+    }
+
+    let newString = currentString.substring(0, i);
+    let leftOverString = currentString.substring(i);
+    newString += replaceValue + leftOverString.substring(leftOverString.indexOf("\n"));
+    if (newString == currentString) {
+        return newString;
+    }
+    return replaceAll(newString, replaceToken, replaceValue);
+}
+
+function apply(configurationPath: string) {
+    var command = tl.tool(tl.which("kubectl"));
+    command.arg("apply");
+    command.arg(getNameSpace());
+    command.arg(["-f", configurationPath]);
+    return command.execSync();
+}
+
+function annotate(configurationPath: string): any {
+    var command = tl.tool(tl.which("kubectl"));
+    command.arg("annotate");
+    command.arg(getNameSpace());
+    command.arg(["-f", configurationPath]);
+    command.arg(`azure-pipelines/execution=${tl.getVariable("Build.BuildNumber")}`)
+    command.arg(`azure-pipelines/pipeline="${tl.getVariable("Build.DefinitionName")}"`)
+    command.arg(`azure-pipelines/executionuri=${tl.getVariable("System.TeamFoundationCollectionUri")}_build/results?buildId=${tl.getVariable("Build.BuildId")}`)
+    command.arg(`azure-pipelines/project=${tl.getVariable("System.TeamProject")}`)
+    command.arg(`azure-pipelines/org=${tl.getVariable("System.CollectionId")}`)
+    command.arg(`--overwrite`)
+    return command.execSync();
+}
+
+function annotateResourceSets(type, name) {
+    if (type.indexOf("pod") > -1) {
+        return;
+    }
+    function annotatePod(name) {
+        var command = tl.tool(tl.which("kubectl"));
+        command.arg("annotate");
+        command.arg(getNameSpace());
+        command.arg(["pod", name]);
+        command.arg(`azure-pipelines/execution=${tl.getVariable("Build.BuildNumber")}`)
+        command.arg(`azure-pipelines/pipeline="${tl.getVariable("Build.DefinitionName")}"`)
+        command.arg(`azure-pipelines/executionuri=${tl.getVariable("System.TeamFoundationCollectionUri")}_build/results?buildId=${tl.getVariable("Build.BuildId")}`)
+        command.arg(`azure-pipelines/project=${tl.getVariable("System.TeamProject")}`)
+        command.arg(`azure-pipelines/org=${tl.getVariable("System.CollectionId")}`)
+        command.arg(`--overwrite`)
+        return command.execSync();
+    }
+
+    var owner = name;
+    if (type.indexOf("deployment") > -1) {
+        owner = getNewReplicaSet(name);
+    }
+
+    var allPods = JSON.parse(getAllPods().stdout);
+    allPods["items"].forEach((pod) => {
+        let owners = pod["metadata"]["ownerReferences"];
+        owners.forEach(ownerRef => {
+            if (ownerRef["name"] == owner) {
+                annotatePod(pod["metadata"]["name"]);
+            }
+        });
+    });
+}
+
+function getNewReplicaSet(deployment) {
```

Comment:
Ideally all such helper functions should be in a OM on top of kubectl so that we dont have N copies of same code lying around.